hack/lib/version.sh

```diff
@@ -89,7 +89,7 @@ kube::version::get_version_vars() {
       # Try to match the "git describe" output to a regex to try to extract
       # the "major" and "minor" versions and whether this is the exact tagged
       # version or whether the tree is between two tagged versions.
-      if [[ "${KUBE_GIT_VERSION}" =~ ^v([0-9]+)\.([0-9]+)(\.[0-9]+)?([-].*)?$ ]]; then
+      if [[ "${KUBE_GIT_VERSION}" =~ ^v([0-9]+)\.([0-9]+)(\.[0-9]+)?([-].*)?([+].*)?$ ]]; then
```

Comment:
As said in https://semver.org/

> Identifiers MUST comprise only ASCII alphanumerics and hyphen [0-9A-Za-z-]

Reply:
This regex is not a very good representation for semver. The fix just allows metadata to be attached to the version to allow the minor and major versions to be parsed correctly. A separate PR should be created, IMO, to fix any issues with the regex itself.